src/agent/src/image_rpc.rs

```diff
@@ -206,6 +208,36 @@ impl ImageService {
             verify_cid(&cid)?;
         }
 
+        // Can switch to use cid directly when we remove umoci
+        let v: Vec<&str> = image.rsplit('/').collect();
+        if !v[0].is_empty() && v[0].starts_with("pause:") {
+            let cc_pause_bundle = Path::new(KATA_CC_PAUSE_BUNDLE);
+            if !cc_pause_bundle.exists() {
+                return Err(anyhow!("Pause image not present in rootfs"));
+            }
+
+            info!(sl!(), "use guest pause image cid {:?}", cid);
+            let pause_bundle = Path::new(CONTAINER_BASE).join(&cid);
+            let pause_rootfs = pause_bundle.join("rootfs");
+            let pause_config = pause_bundle.join(CONFIG_JSON);
+            let pause_binary = pause_rootfs.join("pause");
+            tokio::fs::create_dir_all(&pause_rootfs).await?;
+            if !pause_config.exists() {
+                tokio::fs::copy(
+                    cc_pause_bundle.join(CONFIG_JSON),
+                    pause_bundle.join(CONFIG_JSON),
+                )
+                .await?;
+            }
+            if !pause_binary.exists() {
+                tokio::fs::copy(cc_pause_bundle.join("rootfs").join("pause"), pause_binary).await?;
+            }
+
+            let mut sandbox = self.sandbox.lock().await;
+            sandbox.images.insert(String::from(image), cid.to_string());
+            return Ok(image.to_owned());
+        }
+
         if !aa_kbc_params.is_empty() {
```

Comment:
I suppose it's intentional to not start the AA when we unpack the pause image?

Reply:
Yes, pull `pause` image don't need start AA and `PullImageRequest` will pull one image each time, AA will be started when pull other images if needed.